src/fsharp/IlxGen.fs

```diff
@@ -6130,6 +6142,13 @@ and GenMethodForBinding
          ctorThisValOpt, baseValOpt, methLambdaTypars, methLambdaVars, methLambdaBody, returnTy) =
     let g = cenv.g
     let m = v.Range
+
+    // When emitting a reference assembly, do not emit methods that are private unless they are virtual/abstract or provide an explicit interface implementation.
+    // Internal methods can be omitted only if the assembly does not contain a System.Runtime.CompilerServices.InternalsVisibleToAttribute.
+    if cenv.opts.referenceAssemblyOnly && 
+       (v.Accessibility.IsPrivate || (v.Accessibility.IsInternal && not cenv.hasInternalsVisibleToAttr)) && 
```

Comment:
The `Accessibility` thing on Val and Entity is not complete unfortunately and most code relying on it alone is probably buggy.  See https://github.com/dotnet/fsharp/issues/11528

Reply:
We'll be using ILMemberAccess here as it's the most accurate, so we don't have to do any extra work for accessibility. The only downside is that the compiler will always emit Internal for everything and never private. I'm wondering if we could change that in the future. Logically speaking, there are things that really should be ILMemberAccess.Private when emitting.